File: M3Tools/Dialogs/Emails/CustomEmailDialog.vb
Imports System.Windows.Forms

Public Class CustomEmailDialog
	Public ReadOnly Property Email As CustomEmail
		Get
			Return ce_Email
		End Get
	End Property

	ReadOnly Property Subject As String
		Get
			Return Email.Subject
		End Get
	End Property

	ReadOnly Property Body As String
		Get
			Return Email.Body
		End Get
	End Property

	ReadOnly Property RTBody As String
		Get
			Return Email.RichTextBody
		End Get
	End Property

	Private Sub FinishedCustom(sender As Object, e As EventArgs) Handles btn_Done.Click
		Me.DialogResult = DialogResult.OK
		Me.Close()
	End Sub

	Private Sub CancelCustom(sender As Object, e As EventArgs) Handles Cancel_Button.Click
		Me.DialogResult = DialogResult.Cancel
		Me.Close()
	End Sub

End Class

File: M3Tools/Dialogs/Emails/CustomEmailDialog.cs
using System;
using System.Windows.Forms;
using SPPBC.M3Tools.Types.Extensions;

namespace SPPBC.M3Tools
{
	/// <summary>
	/// The dialog for creating a new email
	/// </summary>
    public partial class CustomEmailDialog
    {
		/// <summary>
		/// The subject of the email
		/// </summary>
        public string Subject
        {
            get
            {
                return ce_Email.Subject;
            }
        }

        private string Body
        {
            get
            {

                return ce_Email.Body;
            }
        }

		/// <summary>
		/// 
		/// </summary>
        public CustomEmailDialog()
        {
            InitializeComponent();
        }

        private void FinishedCustom(object sender, EventArgs e)
        {
            DialogResult = DialogResult.OK;
            Close();
        }

        private void CancelCustom(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
            Close();
        }

    }
}
